spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -652,18 +568,14 @@ private void configureOutboundHandler(AmqpOutboundEndpoint handler, RabbitProper
 	}
 
 	@Override
-	public void bindPubSubProducer(String name, MessageChannel moduleOutputChannel,
-			Properties properties) {
-		validateProducerProperties(name, properties, SUPPORTED_PUBSUB_PRODUCER_PROPERTIES);
+	public void bindProducer(String name, MessageChannel outputChannel, Properties properties) {
+		validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
 		RabbitPropertiesAccessor accessor = new RabbitPropertiesAccessor(properties);
 		String exchangeName = applyPrefix(accessor.getPrefix(this.defaultPrefix), name);
```

Comment:
It looks like the method invocation `buildOutboundEndpoint` also declares the `exchange` with the same name and the steps at lines 574:576 can be removed?